src/Service/OpenBelastingService.php

```diff
@@ -29,30 +32,220 @@ class OpenBelastingService
      */
     private EntityManagerInterface $entityManager;
 
+    /**
+     * @var MappingService
+     */
+    private MappingService $mappingService;
+
     /**
      * The plugin logger.
      *
      * @var LoggerInterface
      */
     private LoggerInterface $logger;
 
+    /**
+     * @var Mapping|null The mapping we are using for the outgoing call.
+     */
+    private ?Mapping $mapping;
+
 
     /**
-     * @param EntityManagerInterface $entityManager The Entity Manager.
-     * @param LoggerInterface        $pluginLogger  The plugin version of the logger interface.
+     * @param EntityManagerInterface $entityManager  The Entity Manager.
+     * @param LoggerInterface        $pluginLogger   The plugin version of the logger interface.
+     * @param MappingService         $mappingService MappingService.
      */
     public function __construct(
         EntityManagerInterface $entityManager,
-        LoggerInterface $pluginLogger
+        LoggerInterface $pluginLogger,
+        MappingService $mappingService
     ) {
-        $this->entityManager = $entityManager;
-        $this->logger        = $pluginLogger;
-        $this->configuration = [];
-        $this->data          = [];
+        $this->entityManager  = $entityManager;
+        $this->logger         = $pluginLogger;
+        $this->mappingService = $mappingService;
+        $this->configuration  = [];
+        $this->data           = [];
 
     }//end __construct()
 
 
+    /**
+     * Gets and sets a Mapping object using the required configuration['mapping'] to find the correct Mapping.
+     *
+     * @return Mapping|null The Mapping object we found or null if we don't find one.
+     */
+    private function setMapping(): ?Mapping
+    {
+        $this->mapping = $this->entityManager->getRepository('App:Mapping')->findOneBy(['reference' => $this->configuration['mapping']]);
+        if ($this->mapping instanceof Mapping === false) {
+            $this->logger->error("No mapping found with reference: {$this->configuration['mapping']}");
+
+            return null;
+        }
+
+        return $this->mapping;
+
+    }//end setMapping()
```

Comment:
Instead of this function try and use:
https://github.com/CommonGateway/CoreBundle/blob/99dba7f8263b33851edeb291ee3952d17530935a/Service/GatewayResourceService.php#L74 

Here is an example:
https://github.com/OpenCatalogi/OpenCatalogiBundle/blob/a14689bdee3087d457e3cbbec717b5e40315bef4/Service/GithubEventService.php#L171

Reply:
You might want to do:
```
$mapping = $this->resourceService->getMapping('https://api.github.com/oc.githubRepository.mapping.json', 'open-catalogi/open-catalogi-bundle');
if $mapping === null { return [] }
```
however